packages/@aws-cdk/cdk/lib/cfn-resource.ts

```diff
@@ -198,8 +199,8 @@ export class CfnResource extends CfnRefElement {
           // Post-Resolve operation since otherwise deepMerge is going to mix values into
           // the Token objects returned by ignoreEmpty.
           [this.logicalId]: new PostResolveToken({
-            Type: this.resourceType,
-            Properties: ignoreEmpty(this.renderProperties()),
+            Type: this.cfnResourceType,
+            Properties: ignoreEmpty(this.properties),
```

Comment:
this.properties is now a getter. Having this inside the token forces resolve before renderProperties which is needed for validators that run in the render function. @rix0rrr I could refactor, but now that all render does is validate the resolved properties, this feels ok?